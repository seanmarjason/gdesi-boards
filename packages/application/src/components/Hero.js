import * as React from 'react';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';
import InputLabel from '@mui/material/InputLabel';
import Link from '@mui/material/Link';
import Stack from '@mui/material/Stack';
import TextField from '@mui/material/TextField';
import Typography from '@mui/material/Typography';
import visuallyHidden from '@mui/utils/visuallyHidden';

import NextLink from 'next/link';


export default function Hero() {
  return (
    <Box
      id="hero"
      sx={(theme) => ({
        width: '100%',
        backgroundRepeat: 'no-repeat',
        backgroundImage:
          'radial-gradient(ellipse 80% 50% at 50% -20%, hsl(210, 100%, 90%), transparent)',
        ...theme.applyStyles('dark', {
          backgroundImage:
            'radial-gradient(ellipse 80% 50% at 50% -20%, hsl(210, 100%, 16%), transparent)',
        }),
      })}
    >
      <Container
        sx={{
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          pt: { xs: 14, sm: 20 },
          pb: { xs: 8, sm: 12 },
        }}
      >
        <Stack
          spacing={2}
          useFlexGap
          sx={{ alignItems: 'center', width: { xs: '100%', sm: '70%' } }}
        >
          <Typography
            variant="h1"
            sx={{
              display: 'flex',
              flexDirection: { xs: 'column', sm: 'row' },
              alignItems: 'center',
              textAlign: 'center',
              fontSize: 'clamp(3rem, 10vw, 3.5rem)',
            }}
          >
            Remote & Hybrid Working Done Right!
          </Typography>
          <Typography
            sx={{
              textAlign: 'center',
              color: 'text.secondary',
              width: { sm: '100%', md: '80%' },
            }}
          >
            Explore our latest tools to support your remote or
            hybrid team to collaborate across borders and timezones.
          </Typography>
          <Box sx={{gap: 4}}>
          <Button component={NextLink} href="/sign-up" variant="contained" color="primary">
            Register
          </Button>
          <Button onClick={() => signIn()} variant="contained" color="secondary">
            Sign in
          </Button>
          </Box>
        </Stack>
        {/* <StyledBox id="image" /> */}
      </Container>
    </Box>
  );
}
