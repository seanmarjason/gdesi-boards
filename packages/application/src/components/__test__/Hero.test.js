import { render } from '@testing-library/react'
import Hero from '../Hero';

it('Hero renders correctly', () => {
  const { container } = render(<Hero />)
  expect(container).toMatchSnapshot()
})
