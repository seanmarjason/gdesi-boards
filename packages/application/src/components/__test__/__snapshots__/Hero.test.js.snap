// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1ge4665"
    id="hero"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1f9chyi-MuiContainer-root"
    >
      <div
        class="MuiStack-root css-i5068v-MuiStack-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1rdt02p-MuiTypography-root"
        >
          Our latest 
          <span
            class="MuiTypography-root MuiTypography-h1 css-1w0kz0g-MuiTypography-root"
          >
            products
          </span>
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pp1339-MuiTypography-root"
        >
          Explore our cutting-edge dashboard, delivering high-quality solutions tailored to your needs. Elevate your experience with top-tier features and services.
        </p>
        <div
          class="MuiStack-root css-oo7k1y-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-10ple52-MuiFormLabel-root-MuiInputLabel-root"
            for="email-hero"
          >
            Email
          </label>
          <div
            aria-label="Enter your email address"
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-hiddenLabel css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                aria-label="Enter your email address"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputHiddenLabel css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                id="email-hero"
                placeholder="Your email address"
                type="text"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-aq79df-MuiButtonBase-root-MuiButton-root"
            id=":r1:"
            tabindex="0"
            type="button"
          >
            Start now
          </button>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption css-kfgbda-MuiTypography-root"
        >
          By clicking "Start now" you agree to our 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="#"
          >
            Terms & Conditions
          </a>
          .
        </span>
      </div>
    </div>
  </div>
</div>
`;
